test/core/security/grpc_tls_crl_provider_test.cc

```diff
@@ -83,6 +101,110 @@ TEST(CrlProviderTest, StaticCrlProviderLookupBad) {
   EXPECT_EQ(crl, nullptr);
 }
 
+TEST(CrlProviderTest, DirectoryReloaderCrlLookupGood) {
+  auto result = experimental::CreateDirectoryReloaderProvider(
+      kCrlDirectory, std::chrono::seconds(1), nullptr);
+  ASSERT_TRUE(result.ok());
+  std::shared_ptr<CrlProvider> provider = std::move(*result);
+
+  CertificateInfoImpl cert = CertificateInfoImpl(kCrlIssuer);
+  auto crl = provider->GetCrl(cert);
+  ASSERT_NE(crl, nullptr);
+  ASSERT_EQ(crl->Issuer(), kCrlIssuer);
+
+  CertificateInfoImpl intermediate =
+      CertificateInfoImpl(kCrlIntermediateIssuer);
+  auto intermediate_crl = provider->GetCrl(intermediate);
+  ASSERT_NE(intermediate_crl, nullptr);
+  ASSERT_EQ(intermediate_crl->Issuer(), kCrlIntermediateIssuer);
+}
+
+TEST(CrlProviderTest, DirectoryReloaderCrlLookupBad) {
```

Comment:
Suggest calling this `IssuerNotFound` instead of `Bad`.

Reply:
Changed

Reply:
Maybe this test name should be "LookupMalformedCert"?

Reply:
Changed to `MissingIssuer` - we don't really have a malformed cert because we are just creating `CertInfo` object and populating it with an issuer that doesn't exist